src/SignalR/clients/csharp/Client.SourceGenerator/src/gen/Microsoft.AspNetCore.SignalR.Client.SourceGenerator.csproj

```diff
@@ -1,19 +1,25 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
+    <Description>Source Generator for ASP.NET Core SignalR Client</Description>
     <TargetFramework>netstandard2.0</TargetFramework>
-    <LangVersion>9</LangVersion>
     <IncludeBuildOutput>false</IncludeBuildOutput>
-    <SuppressDependenciesWhenPacking>true</SuppressDependenciesWhenPacking>
+    <SuppressDependenciesWhenPacking>false</SuppressDependenciesWhenPacking>
```

Comment:
Was this intended? The only non-private asset I see is HubProxyAttributes which isn't packaged.

![image](https://user-images.githubusercontent.com/54385/139933046-369b771f-71c4-45de-8575-462fe3ce3546.png)

Reply:
It's needed to put the .netstandard2.0 framework in the nuspec. Changed the reference to not bring in HubProxyAttributes as a reference.

Reply:
`dotnet pack` now fails for me with the following: 

```
C:\dev\dotnet\aspnetcore\.dotnet\sdk\7.0.100-alpha.1.21551.3\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5019: File not found: 'C:\dev\dotnet\aspnetcore\artifacts\bin\Microsoft.AspNetCore.SignalR.Client.SourceGenerator\Debug\netstandard2.0\Microsoft.AspNetCore.SignalR.Client.SourceGenerator.HubProxyAttributes.dll'. [C:\dev\dotnet\aspnetcore\src\SignalR\clients\csharp\Client.SourceGenerator\src\gen\Microsoft.AspNetCore.SignalR.Client.SourceGenerator.csproj]
```

What still seems to work for me is `PrivateAssets="all"` but then you also need a supression with `<NoWarn>BUILD005;$(NoWarn)</NoWarn>` or similar to silence a warning about using `PrivateAssets` with project references. It seems to do the right thing though.

Reply:
`PrivateAssets="all"` shouldn't work w/ project references. `Private="true"` is the way to go.

Reply:
I tried `Private="true"` since that's what the build warning suggests, but that still leaves "Microsoft.AspNetCore.SignalR.Client.SourceGenerator.HubProxyAttributes" as a dependency in the nupkg.

Reply:
Ah, I meant to `Private="false"` but that controls copy-local and not dependencies added to a package. It wouldn't work alone (but see below).

@rainersigwald why does `PrivateAssets="all"` seem to work in this case❔ It's documented as applicable only for `@(PackageReference)` items and line 16 turns into a `@(ProjectReference)`.

@BrennanConroy @halter73 another option: Use all of

``` msbuild
      Private="false"
      ReferenceOutputAssembly="false"
      SkipGetTargetFrameworkProperties="true"
```

And, change the `@(None)` item to copy from `$(ArtifactsDir)bin\HubProxyAttributes\$(Configuration)\netstandard2.0`.

For one thing, this option would avoid timing issues due to `HubProxyAttributes` holding a file lock a bit after its build completes. `@(None)` items are handled (copied to this project's bin/ folder) a bit later than copy-local assemblies IIRC. `ReferenceOutputAssembly="false"` ensures NuGet and most of MSBuild ignores the referenced project and its assembly.

Reply:
> @rainersigwald why does PrivateAssets="all" seem to work in this case❔ It's documented as applicable only for @(PackageReference) items and line 16 turns into a @(ProjectReference).

My guess is that it's inherited by an item generated by NuGet or the SDK flattening references and bringing transitive ones to this project, but I don't know for sure.